client/src/backend/ContractsAPI.ts

```diff
@@ -122,74 +121,34 @@ export class ContractsAPI extends EventEmitter {
     this.emit(ContractsAPIEvent.TxSubmitted, txDiagnosticInfo);
   }
 
-  public destroy(): void {
-    this.removeEventListeners();
-  }
-
   private makeCall<T>(contractViewFunction: ContractFunction<T>, args: unknown[] = []): Promise<T> {
     return this.contractCaller.makeCall(contractViewFunction, args);
   }
 
-  public async setupEventListeners(): Promise<void> {
-    const { coreContract } = this;
-
-    const filter = {
-      address: coreContract.address,
-      topics: [
-        [
-          coreContract.filters.PlayerUpdated(null, null).topics,
-          coreContract.filters.BattleUpdated(null, null).topics,
-        ].map((topicsOrUndefined) => (topicsOrUndefined || [])[0]),
-      ] as Array<string | Array<string>>,
-    };
-
-    const eventHandlers = {
-      [ContractEvent.PlayerUpdated]: (
-        rawAddress: string,
-        commitment: BigNumber,
-        blockNum: BigNumber
-      ) => {
-        this.emit(
-          ContractsAPIEvent.PlayerUpdated,
-          address(rawAddress),
-          commitment.toBigInt(),
-          blockNum.toBigInt()
-        );
-      },
-      [ContractEvent.BattleUpdated]: (player1: string, player2: string) => {
-        this.emit(ContractsAPIEvent.BattleUpdated, address(player1), address(player2));
-      },
-    };
-
-    this.ethConnection.subscribeToContractEvents(coreContract, eventHandlers, filter);
-  }
-
-  public removeEventListeners(): void {
-    const { coreContract } = this;
-
-    coreContract.removeAllListeners(ContractEvent.PlayerUpdated);
-    coreContract.removeAllListeners(ContractEvent.BattleUpdated);
-  }
-
   public async getGridUpperBound(): Promise<number> {
-    return (await this.makeCall<EthersBN>(this.coreContract.GRID_UPPER_BOUND)).toNumber();
+    const contract = await this.getContract();
+    return (await this.makeCall<EthersBN>(contract.GRID_UPPER_BOUND)).toNumber();
   }
 
   public async getSaltUpperBound(): Promise<number> {
-    return (await this.makeCall<EthersBN>(this.coreContract.SALT_UPPER_BOUND)).toNumber();
+    const contract = await this.getContract();
+    return (await this.makeCall<EthersBN>(contract.SALT_UPPER_BOUND)).toNumber();
   }
 
   public async getWins(address: EthAddress): Promise<number> {
-    return (await this.makeCall<EthersBN>(this.coreContract.getWins, [address])).toNumber();
+    const contract = await this.getContract();
```

Comment:
dumb q, isn't this pattern slower per call? Should there be a cache for the loading in the network package?

Reply:
`loadContract` actually already has a cache, but in looking for the link, I noticed that it is always skipped if the loader is provided, so I'll fix that in the package

Reply:
New implementation of load contract:
```typescript
async loadContract<C extends BaseContract>(address: string, loader?: ContractLoader<C>): Promise<C> {
    const contract = this.#contracts.get(address);
    if (contract) {
      return contract as C;
    } else {
      if (loader) {
        this.#loaders.set(address, loader);
        const contract = await loader(address, this.#connection.provider, this.#connection.signer);
        this.#contracts.set(address, contract);
        return contract as C;
      } else {
        throw new Error(`Contract never loaded. Address: ${address}`);
      }
    }
}
```